src/components/ftbot/PairlistConfigurator.vue

```diff
@@ -0,0 +1,258 @@
+              <i-mdi-arrow-right-bold-box-outline class="fs-4" />
+            </b-button>
+          </b-list-group-item>
+        </b-list-group>
+      </b-col>
+      <b-col>
+        <b-row>
+          <b-col>
+            <b-form-input
+              v-model="config.name"
+              class="mb-2"
+              placeholder="Configuration name..."
+            ></b-form-input>
+          </b-col>
+          <b-col cols="auto">
+            <b-button @click="save">Save</b-button>
+          </b-col>
+        </b-row>
+        <div ref="pairlistConfigsEl" class="h-100">
+          <b-alert
+            :model-value="config.pairlists.length > 0 && !firstPairlistIsGenerator"
+            variant="danger"
+          >
+            First entry in the pairlist must be a Generating pairlist, like StaticPairList or
+            VolumePairList.
+          </b-alert>
+          <PairlistConfigItem
+            v-for="(pairlist, i) in pairlistsComp"
+            :key="pairlist.id"
+            v-model="config.pairlists[i]"
+            :index="i"
+            @remove="removeFromConfig"
+          />
+          <b-button
+            :disabled="evaluating || !pairlistValid"
+            variant="primary"
+            size="lg"
+            squared
+            class="mt-2 evaluate"
+            @click="test"
+            >Evaluate</b-button
+          >
+        </div>
+      </b-col>
+      <b-col cols="12" md="3">
+        <pre class="text-start"><code>{{ configJSON }}</code></pre>
+      </b-col>
+    </b-row>
+  </b-container>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, ref, toRaw, watch } from 'vue';
+import { useBotStore } from '@/stores/ftbotwrapper';
+import PairlistConfigItem from './PairlistConfigItem.vue';
+import { Pairlist, PairlistConfig, PairlistParamType, PairlistPayloadItem } from '@/types';
+import { useSortable, moveArrayElement } from '@vueuse/integrations/useSortable';
+
+const emit = defineEmits([
+  'update:modelValue',
+  'saveConfig',
+  'started',
+  'progress',
+  'done',
+  'error',
+]);
+const props = defineProps<{
+  selectedConfig: PairlistConfig;
+}>();
+
+const botStore = useBotStore();
+
+const availablePairlists = ref<Pairlist[]>([]);
+const config = ref<PairlistConfig>(props.selectedConfig);
+const pairlistConfigsEl = ref<HTMLElement | null>(null);
+const availablePairlistsEl = ref<HTMLElement | null>(null);
+const evaluating = ref(false);
+
+const firstPairlistIsGenerator = computed<boolean>(() => {
+  // First pairlist must be a generator
+  if (config.value.pairlists[0]?.is_pairlist_generator) {
+    return true;
+  }
+  return false;
+});
+
+const pairlistValid = computed<boolean>(() => {
+  return firstPairlistIsGenerator.value && config.value.pairlists.length > 0;
+});
+
+// v-for updates with sorting, deleting and adding items seem to get wonky without unique keys for every item
+const pairlistsComp = computed(() =>
+  config.value.pairlists.map((p) => {
+    if (p.id) {
```

Comment:
I'm not entirely sure why - but this throws an error when reordering the pairlist.

To reproduce:
* start with a pairlist of length 2
* Drag the 1st pairlist to the 2nd place (this works)
* drag the 2nd pairlist to the top (this throws an error, as the list now seems to contain 3 items, with "undefined" being the content of the 2nd list.

my best guess for now is that it's an incompatibility between useSortable and using a computed property for the actual display - but that's just a guess for now.

Reply:
I can reproduce it when one isn't a generator, seems to work fine otherwise. Seems like the inline alert is messing something up?